Extensions/3D/Model3DRuntimeObject3DRenderer.ts

```diff
@@ -67,34 +69,71 @@ namespace gdjs {
      * The 3D model stretched in a 1x1x1 cube.
      */
     private _threeObject: THREE.Object3D;
+    private _originalModel: THREE_ADDONS.GLTF;
+    private _animationMixer: THREE.AnimationMixer;
+    private _action: THREE.AnimationAction | null;
+
+    private _modelOriginPoint: FloatPoint3D;
 
     constructor(
       runtimeObject: gdjs.Model3DRuntimeObject,
       instanceContainer: gdjs.RuntimeInstanceContainer
     ) {
-      // @ts-ignore It can't be null if THREE exists.
-      const originalModelMesh: THREE.Object3D = instanceContainer
+      // GLB files with skeleton must not have any transformation to work properly.
+      const originalModel = instanceContainer
         .getGame()
         .getModel3DManager()
         .getModel(runtimeObject._modelResourceName);
-      const modelObject3D = THREE_ADDONS.SkeletonUtils.clone(originalModelMesh);
+      // _updateModel will actually add a clone of the model.
+      const model = new THREE.Group();
 
       // Create a group to transform the object according to
       // position, angle and dimensions.
       const group = new THREE.Group();
       group.rotation.order = 'ZYX';
-      group.add(modelObject3D);
+      group.add(model);
       super(runtimeObject, instanceContainer, group);
 
       this._model3DRuntimeObject = runtimeObject;
-      this._threeObject = modelObject3D;
+      this._threeObject = model;
+      this._originalModel = originalModel;
+      this._modelOriginPoint = [0, 0, 0];
 
       this.updateSize();
       this.updatePosition();
       this.updateRotation();
+
+      this._animationMixer = new THREE.AnimationMixer(model);
+      this._action = null;
+    }
+
+    updateAnimation(timeDelta: float) {
+      this._animationMixer.update(timeDelta);
+    }
+
+    updatePosition() {
+      const originPoint = this.getOriginPoint();
+      const centerPoint = this.getCenterPoint();
+      this.get3DRendererObject().position.set(
+        this._object.getX() -
+          this._object.getWidth() * (originPoint[0] - centerPoint[0]),
+        this._object.getY() -
+          this._object.getHeight() * (originPoint[1] - centerPoint[1]),
+        this._object.getZ() -
+          this._object.getDepth() * (originPoint[2] - centerPoint[2])
+      );
+    }
+
+    getOriginPoint() {
+      return this._model3DRuntimeObject._originPoint || this._modelOriginPoint;
     }
 
-    _updateDefaultTransformation(
+    getCenterPoint() {
+      return this._model3DRuntimeObject._centerPoint || this._modelOriginPoint;
```

Comment:
Question: why is the fallback needed?

Reply:
It's because null mean the model origin point. It's depend on the object configuration so it can't be set into `_centerPoint` directly.
I added documentation on the 3 attributes.